src/SSHDebugPS/SSH/SSHConnection.cs

```diff
@@ -172,7 +172,8 @@ public override void Close()
 
             if (_remoteSystem != null)
             {
-                _remoteSystem.Dispose();
+                if (_remoteSystem.IsConnected)
+                    _remoteSystem.Disconnect();
```

Comment:
I checked liblinux, and the implementation of `IRemoteSystem.Dispose` just calls `Disconnect`, so if you're seeing a hang when calling `Dispose`, I'd expect to see it when calling `Disconnect` as well.  If checking `IsConnected` first avoids the problem, I'd leave this as a call to `Dispose`, just to maintain the pattern that `IDisposable` objects should be disposed.

Reply:
I sent them email to find out what i should be doing